src/Analyser/NodeScopeResolver.php

```diff
@@ -352,7 +362,29 @@ private function enterForeach(Scope $scope, Foreach_ $node): Scope
 		}
 
 		if ($node->valueVar instanceof List_ || $node->valueVar instanceof Array_) {
-			$scope = $this->lookForArrayDestructuringArray($scope, $node->valueVar);
+			$exprType = $scope->getType($node->expr);
```

Comment:
@ondrejmirtes Maybe you know a better solution how to detect union of array values...